### DISCUSSION TITLE
希望能够支持多个互斥组

### DISCUSSION THREAD

**Discussion by jilianwang** [Tencent/bk-ci#4467] [Ideas]
**What would you like to be added**:
希望能够支持多个互斥组
**Why is this needed**:
当前一个蓝盾stage只能输入一个互斥组。当b、c都要与a互斥的时候只能输入相同互斥组名，使得a、b、c三者互斥组名相同。然而b和c其实并不需要互斥。

**Comment by iccengan** [comment]
@jilianwang 具体什么场景下需要a和b互斥、a和c互斥，但b和c不互斥吗？他们用的是否同一构建资源？

**Comment by jilianwang** [comment]
a是引擎流水线的svn上传组，b和c是两条构建线的svn下载组。如果a在上传，bc就等他传完再下载。如果单一互斥会导致bc不能同时下载

**Comment by iccengan** [comment]
三个Job是在同一条流水线的同个Stage下吗？是的话，新特性Depends On可以满足你的需求：#111
如下图，Job 2-2 和 Job 2-3 都配置了依赖于 Job 2-1。那么只有当 Job 2-1 执行完成，Job 2-2 和 Job 2-3 才会开始执行。
![image](https://user-images.githubusercontent.com/10875468/93743321-7168c280-fc22-11ea-8e71-fc41bfe76ebf.png)
![image](https://user-images.githubusercontent.com/10875468/93743388-8fcebe00-fc22-11ea-8966-080e0c7951ac.png)

这个特性预计9月24日发至灰度环境，下一个版本日发至生产（预计国庆后）。

**Comment by gamebaker** [comment]
我们也有类似的需求：
A和B、A和C分别互斥；
但是B、C之间不互斥；
这就需要A分别设置2个互斥组。

depends on无法满足需求：
首先A、B、C是跨流水线的，depends on只能同流水线；
其次B、C依赖A，我们有需求A也要依赖B、C（其实就是互斥），即B、C有任意一个在运行，A也不能运行。

**Reply by zanyzhao** [reply]
如果ABC这么强互斥，可否评估下把ABC放在同一个Stage的可行性呢？
